### PR TITLE
Use const rather than var when target es6

### PR DESCRIPTION
**Pull request by plantain-00**
#6092

### CODE REVIEW

File: src/compiler/emitter.ts [hunk 1]
```diff
@@ -6146,7 +6158,12 @@ var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, ge
                         }
                         else if (namespaceDeclaration && isDefaultImport(node)) {
                             // import d, * as x from "foo"
-                            write("var ");
+                            if (languageVersion !== ScriptTarget.ES6) {
```
Comment (DanielRosenwasser) [inline src/compiler/emitter.ts]: `<= ScriptTarget.ES5`

File: src/compiler/emitter.ts [hunk 2]
```diff
@@ -6107,7 +6107,14 @@ var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, ge
                         if (namespaceDeclaration && !isDefaultImport(node)) {
                             // import x = require("foo")
                             // import * as x from "foo"
-                            if (!isExportedImport) write("var ");
+                            if (!isExportedImport) {
+                                if (languageVersion !== ScriptTarget.ES6) {
```
Comment (DanielRosenwasser) [inline src/compiler/emitter.ts]: `<= ScriptTarget.ES5`

File: src/compiler/emitter.ts (hunk 1 above)
Comment (plantain-00) [inline src/compiler/emitter.ts]: done

File: src/compiler/emitter.ts (hunk 2 above)
Comment (plantain-00) [inline src/compiler/emitter.ts]: done

File: src/compiler/emitter.ts [hunk 3]
```diff
@@ -6107,7 +6107,14 @@ var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, ge
                         if (namespaceDeclaration && !isDefaultImport(node)) {
                             // import x = require("foo")
                             // import * as x from "foo"
-                            if (!isExportedImport) write("var ");
+                            if (!isExportedImport) {
+                                if (languageVersion <= ScriptTarget.ES5) {
```
Comment (paulvanbrenk) [inline src/compiler/emitter.ts]: Can you wrap this in a helper method, since you repeat this pattern 3 times in this PR

Comment (plantain-00) [inline src/compiler/emitter.ts]: done

File: src/compiler/emitter.ts [hunk 4]
```diff
@@ -6107,7 +6116,9 @@ var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, ge
                         if (namespaceDeclaration && !isDefaultImport(node)) {
                             // import x = require("foo")
                             // import * as x from "foo"
-                            if (!isExportedImport) write("var ");
+                            if (!isExportedImport) {
```
Comment (RyanCavanaugh) [inline src/compiler/emitter.ts]: The function name (`emitVar`) is misleading and we probably don't need the function at all.

I'd prefer to just have a local in this function `const varOrConst = (languageVersion <= ScriptTarget.ES5) ? 'var ' : 'const ';`) and `write(varOrConst)` at these call sites.

Comment (plantain-00) [inline src/compiler/emitter.ts]: done

**Comment by RyanCavanaugh** [comment]
:+1:

**Comment by paulvanbrenk** [comment]
:+1:

**Pull request merged**